### ISSUE TITLE
current trunk ogit-http-clone gets a populated .git, but no other files

### ISSUE THREAD

**Issue by yomimono** [mirage/ocaml-git#294]
I tried to clone a repository via smart HTTP.  It looks like it works, but none of the files arrive, although metadata is present:

```
🐫  23.06 ocamldev:/tmp$ ogit-http-clone https://github.com/yomimono/example-unikernels.git
🐫  23.07 ocamldev:/tmp$ echo $?
0
🐫  23.07 ocamldev:/tmp$ cd example-unikernels/
🐫  (master) 23.07 ocamldev:/tmp/example-unikernels$ ls
🐫  (master) 23.07 ocamldev:/tmp/example-unikernels$ git log --oneline|head -5
f7a230a fix it!
84feb43 pre-fix
e425f46 use geoff graham's crawl style instead (thanks\!)
38c9737 preliminary oscon demo
5dbfe85 initial unikernel stolen from mirage-skeleton
🐫  (master) 23.07 ocamldev:/tmp/example-unikernels$ ls -alh
total 24K
drwxr-xr-x  3 user user 4.0K May  6 23:07 .
drwxrwxrwt 17 root root  16K May  6 23:07 ..
drwxr-xr-x  4 user user 4.0K May  6 23:07 .git
```

The `ogit-http-clone` here was obtained with `opam pin --dev-repo git` and `opam pin --dev-repo git-unix`, not a released version.  If this is not expected to work with the current trunk, please feel free to just close the issue :)

**Comment by dinosaure** [comment]
Indeed, this behavior is expected. It's most of because we implemented protocol (clone) before INDEX format (which populate your file-system from `.git`). I will make a PR to fix it!

**Comment by dinosaure** [comment]
Fixed by #295.

**Comment by yomimono** [comment]
Thanks for the fix! :)
